Vineyard/Vineyard/CreateGoalView.swift

```diff
@@ -8,44 +8,45 @@
 import SwiftUI
 
 struct CreateGoalView: View {
-//    @Binding var goals: [String] // Binding to update the goals list
+    //    @Binding var goals: [String] // Binding to update the goals list
     @Environment(GroupsListViewModel.self) private var viewModel
     
+    
+    @Binding var indexOfGoal: Int
     @Binding var goals: [Resolution]
-       
-    @State private var action: String = "" // the name of the goal
-    @State private var description: String = "" // the name of the goal
-    @State private var quantity: String = ""
+    @Binding var editMode: Bool
+    
+    @State var action: String = "" // the name of the goal
```

Comment:
`action` reminds me of a `Button`s action or something similar, we can rename this:

```suggestion
    @State var goalName: String = "" // the name of the goal
```